app/controllers/admin/users_controller.rb

```diff
@@ -17,13 +19,30 @@ class UsersController < ApplicationController
     helper_method :sort_column, :sort_direction
 
     def index
-      # search
-      if params[:q].present?
-        @users = user_scope.text_search(params[:q])
-        @inactive_users = User.inactive.text_search(params[:q])
-      else
-        @users = user_scope
-        @inactive_users = User.inactive
+      # Handle search queries
+      handle_search_query
```

Comment:
avoid double render?
```suggestion
      handle_search_query
      return if performed?
```